File: testers/combined/java/generated-and-checked-in/avro_events/combined/avro_events/consumer/AddressHandler.java
package combined.avro_events.consumer;

import com.example.events.Address;
import combined.avro_events.header.StandardHeaders;
import java.lang.Void;
import java.util.concurrent.CompletableFuture;

/** Handler interface for address topic events */
public interface AddressHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(
    String key,
    Address value,
    StandardHeaders headers
  );
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/AddressHandler.kt
package combined.avro_events.consumer

import com.example.events.Address
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Uni

/** Handler interface for address topic events */
interface AddressHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: Address,
    headers: StandardHeaders
  ): Uni<Unit>
}